units/crons.go

```diff
@@ -45,7 +45,7 @@ func PopulateCatchupJobs(part int) amboy.QueueOperation {
 		for _, proj := range projects {
 			// only do catchup jobs for enabled projects
 			// that track push events.
-			if !proj.Enabled || !proj.TracksPushEvents {
+			if !proj.Enabled || !proj.TracksPushEvents || proj.Owner == "" || proj.Repo == "" || proj.Branch == "" {
```

Comment:
I was actually thinking to put it in the query on line 37 (it would probably be a new query). The other checks here should probably be in the query too

Reply:
I added a find query to filter project refs with unset Owner, Repo, and Branch.